File: package/network/_network.c
#include "_network.h"
#include "../pikascript-lib/PikaStdDevice/pika_hal.h"

/*
STA_IF: int
AP_IF: int
STAT_IDLE: int
STAT_CONNECTING: int
STAT_WRONG_PASSWORD: int
STAT_NO_AP_FOUND: int
STAT_CONNECT_FAIL: int
STAT_GOT_IP: int
*/

void _network___init__(PikaObj* self) {
    obj_setInt(self, "STA_IF", PIKA_HAL_WIFI_MODE_STA);
    obj_setInt(self, "AP_IF", PIKA_HAL_WIFI_MODE_AP);
    obj_setInt(self, "STAT_IDLE", PIKA_HAL_WIFI_STATUS_IDLE);
    obj_setInt(self, "STAT_CONNECTING", PIKA_HAL_WIFI_STATUS_CONNECTING);
    obj_setInt(self, "STAT_WRONG_PASSWORD",
               PIKA_HAL_WIFI_STATUS_WRONG_PASSWORD);
    obj_setInt(self, "STAT_NO_AP_FOUND", PIKA_HAL_WIFI_STATUS_NO_AP_FOUND);
    obj_setInt(self, "STAT_CONNECT_FAIL", PIKA_HAL_WIFI_STATUS_CONNECT_FAIL);
    obj_setInt(self, "STAT_GOT_IP", PIKA_HAL_WIFI_STATUS_GOT_IP);
};

File: package/network/network.py
import _network

STA_IF = _network.STA_IF
AP_IF = _network.AP_IF

STAT_IDLE = _network.STAT_IDLE
STAT_CONNECTING = _network.STAT_CONNECTING
STAT_WRONG_PASSWORD = _network.STAT_WRONG_PASSWORD
STAT_NO_AP_FOUND = _network.STAT_NO_AP_FOUND
STAT_CONNECT_FAIL = _network.STAT_CONNECT_FAIL
STAT_GOT_IP = _network.STAT_GOT_IP

_instence:"WLAN" = None
class WLAN(_network.WLAN):
    def __init__(self, interface_id: int):
        global _instence
        super().__init__(interface_id)
        if _instence is None:
            _instence = self

    def active(self, is_active=None):
        if is_active is None:
            return super().checkActive()
        else:
            return super().active(is_active)

    def connect(self, ssid=None, key=None, bssid=None):
        if bssid is None:
            return super().connect(ssid, key)
        else:
            return super().connectWithBssid(ssid, key, bssid)

    def disconnect(self):
        return super().disconnect()

    def status(self, param=None):
        if param is None:
            return super().status()
        else:
            return super().statusWithParam(param)

    def isconnected(self) -> int:
        return super().isconnected()

    def config(self, *para, **kwargs):
        if len(para) == 1:
            return super().checkConfig(para[0])
        else:
            return super().config(**kwargs)

    def ifconfig(self, config=None):
        if config is None:
            t = super().checkIfconfig()
            return (t[0], t[1], t[2], t[3])
        else:
            return super().ifconfig(
                config[0], config[1], config[2], config[3])

    def scan(self):
        return super().scan()

def isconnected():
    if _instence is None:
        return False
    return _instence.isconnected()
